src/coreclr/src/jit/stacklevelsetter.cpp

```diff
@@ -325,6 +328,11 @@ void StackLevelSetter::CheckArgCnt()
 #endif
         comp->SetInterruptible(false);
     }
+#else
+    assert(comp->compCanEncodePtrArgCntMax());
+#endif
+
+#if defined(TARGET_X86)
     if (maxStackLevel >= sizeof(unsigned))
```

Comment:
this is the actual change, as you can see from the comment it is applicable only to x86 where we push arguments (and use EBP) but it was forgotten to be guarded as such before CoreCLR initial PR.